error: '__COUNTER__' is a C2y extension

```
/__w/votca/votca/tools/src/tests/test_cubicspline.cc:33:1: error: '__COUNTER__' is a C2y extension [-Werror,-Wc2y-extensions]
402
   33 | BOOST_AUTO_TEST_SUITE(cubicspline_test)
403
      | ^
404
/usr/include/boost/test/unit_test_suite.hpp:75:9: note: expanded from macro 'BOOST_AUTO_TEST_SUITE'
405
   75 |         BOOST_AUTO_TEST_SUITE_NO_DECOR,                                 \
406
      |         ^
407
/__w/votca/votca/tools/src/tests/test_cubicspline.cc:35:1: error: '__COUNTER__' is a C2y extension [-Werror,-Wc2y-extensions]
408
   35 | BOOST_AUTO_TEST_CASE(spline_grid_high) {
409
      | ^
410
/usr/include/boost/test/unit_test_suite.hpp:212:9: note: expanded from macro 'BOOST_AUTO_TEST_CASE'
411
  212 |         BOOST_AUTO_TEST_CASE_NO_DECOR,                                  \
412
      |         ^
413
/usr/include/eigen3/Eigen/src/Core/DenseBase.h:94:3: note: unnamed type used in template argument was declared here
414
   94 |   enum {
415
      |   ^
416
/usr/include/eigen3/Eigen/src/Core/DenseBase.h:94:3: note: unnamed type used in template argument was declared here
417
/usr/include/eigen3/Eigen/src/Core/DenseBase.h:94:3: note: unnamed type used in template argument was declared here
418
/usr/include/eigen3/Eigen/src/Core/DenseBase.h:94:3: note: unnamed type used in template argument was declared here
419
/__w/votca/votca/tools/src/tests/test_cubicspline.cc:52:1: error: '__COUNTER__' is a C2y extension [-Werror,-Wc2y-extensions]
420
   52 | BOOST_AUTO_TEST_CASE(spline_grid_low) {
421
      | ^
422
/usr/include/boost/test/unit_test_suite.hpp:212:9: note: expanded from macro 'BOOST_AUTO_TEST_CASE'
423
  212 |         BOOST_AUTO_TEST_CASE_NO_DECOR,                                  \
424
      |         ^
425
/__w/votca/votca/tools/src/tests/test_cubicspline.cc:69:1: error: '__COUNTER__' is a C2y extension [-Werror,-Wc2y-extensions]
426
   69 | BOOST_AUTO_TEST_CASE(cubicspline_fit_test) {
427
      | ^
428
/usr/include/boost/test/unit_test_suite.hpp:212:9: note: expanded from macro 'BOOST_AUTO_TEST_CASE'
429
  212 |         BOOST_AUTO_TEST_CASE_NO_DECOR,                                  \
430
      |         ^
431
/__w/votca/votca/tools/src/tests/test_cubicspline.cc:125:1: error: '__COUNTER__' is a C2y extension [-Werror,-Wc2y-extensions]
432
  125 | BOOST_AUTO_TEST_CASE(cubicspline_interpolate_test) {
433
      | ^
434
/usr/include/boost/test/unit_test_suite.hpp:212:9: note: expanded from macro 'BOOST_AUTO_TEST_CASE'
435
  212 |         BOOST_AUTO_TEST_CASE_NO_DECOR,                                  \
436
      |         ^
437
/__w/votca/votca/tools/src/tests/test_cubicspline.cc:170:1: error: '__COUNTER__' is a C2y extension [-Werror,-Wc2y-extensions]
438
  170 | BOOST_AUTO_TEST_CASE(cubicspline_matrix_test) {
439
      | ^
440
/usr/include/boost/test/unit_test_suite.hpp:212:9: note: expanded from macro 'BOOST_AUTO_TEST_CASE'
441
  212 |         BOOST_AUTO_TEST_CASE_NO_DECOR,                                  \
442
      |         ^
443
/usr/include/eigen3/Eigen/src/Core/DenseBase.h:94:3: note: unnamed type used in template argument was declared here
444
   94 |   enum {
445
      |   ^
446
/usr/include/eigen3/Eigen/src/Core/DenseBase.h:94:3: note: unnamed type used in template argument was declared here
447
/usr/include/eigen3/Eigen/src/Core/DenseBase.h:94:3: note: unnamed type used in template argument was declared here
448
/usr/include/eigen3/Eigen/src/Core/DenseBase.h:94:3: note: unnamed type used in template argument was declared here
449
/__w/votca/votca/tools/src/tests/test_cubicspline.cc:192:1: error: '__COUNTER__' is a C2y extension [-Werror,-Wc2y-extensions]
450
  192 | BOOST_AUTO_TEST_SUITE_END()
451
      | ^
452
/usr/include/boost/test/unit_test_suite.hpp:126:49: note: expanded from macro 'BOOST_AUTO_TEST_SUITE_END'
453
  126 | #define BOOST_AUTO_TEST_SUITE_END()             \
454
      |                                                 ^
455
/usr/include/boost/test/unit_test_suite.hpp:368:68: note: expanded from macro '\
456
BOOST_AUTO_TU_REGISTRAR'
457
  368 | static boost::unit_test::ut_detail::auto_test_unit_registrar       \
458
      |                                                                    ^
459
/usr/include/boost/test/unit_test_suite.hpp:358:45: note: expanded from macro '\
460
BOOST_TEST_APPEND_UNIQUE_ID'
461
  358 |   BOOST_JOIN( BOOST_JOIN( name, __LINE__ ), __COUNTER__)
462
      |                                             ^
463
7 errors generated.
```

@baumeier any ideas?

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -37,6 +37,7 @@ Version 2025.1-dev
 -  add integration tests for xtp (#1220)
 -  CI: docker build add provenance=false (#1225)
 -  unbuffered KMC output on single thread (#1226)
+-  CI: ignore c2y-extensions warning on rawhide (#1228)
 
 Version 2025.1 (released 03.10.25)
 ==================================

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -73,6 +73,10 @@ if(ENABLE_WARNING_FLAGS)
   if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-conversion")
   endif()
+  # ignore c2y-extentions warnings on newer clang, this is mostly in boost
+  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 22)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=c2y-extensions")
+  endif()
   if(CMAKE_CXX_COMPILER_ID STREQUAL "Intel") #only icpc, not icpx
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wcheck -Wno-sign-conversion")
   endif()

